helpdesk_mgmt_activity/models/helpdesk_ticket.py

```diff
@@ -0,0 +1,103 @@
+# Copyright (C) 2024 Cetmix OÜ
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import ast
+
+from odoo import _, api, fields, models
+
+
+class HelpdeskTicket(models.Model):
+    _inherit = "helpdesk.ticket"
+
+    can_create_activity = fields.Boolean(related="team_id.is_set_activity")
+    res_model = fields.Char(string="Source Document Model", index=True)
+    res_id = fields.Integer(string="Source Document", index=True)
+
+    record_ref = fields.Reference(
+        selection="_referenceable_models",
+        compute="_compute_record_ref",
+        inverse="_inverse_record_ref",
+        string="Source Record",
+    )
+    source_activity_type_id = fields.Many2one(comodel_name="mail.activity.type")
+    date_deadline = fields.Date(string="Due Date", default=fields.Date.context_today)
+
+    @api.model
+    def _referenceable_models(self):
+        """Select target model for source document"""
+        model_ids_str = (
+            self.env["ir.config_parameter"]
+            .sudo()
+            .get_param("helpdesk_mgmt_activity.helpdesk_available_model_ids", "[]")
+        )
+        model_ids = ast.literal_eval(model_ids_str)
+        if not model_ids:
+            return []
+        IrModelAccess = self.env["ir.model.access"].with_user(self.env.user.id)
+        available_models = self.env["ir.model"].search_read(
+            [("id", "in", model_ids)], fields=["model", "name"]
+        )
+        return [
+            (model.get("model"), model.get("name"))
+            for model in available_models
+            if IrModelAccess.check(model.get("model"), "read", False)
+        ]
+
+    @api.depends("res_model", "res_id")
+    def _compute_record_ref(self):
+        """Compute Source Document Reference"""
+        for rec in self:
+            if rec.res_model and rec.res_id:
+                try:
+                    self.env[rec.res_model].browse(rec.res_id).check_access_rule("read")
+                    rec.record_ref = "%s,%s" % (
+                        rec.res_model,
+                        rec.res_id,
+                    )
+                except Exception:
+                    rec.record_ref = None
```

Comment:
to reduce complexity in the code, you can do a little refactoring: 
```python
for rec in self:
    if not rec.res_model or not rec.res_id:
        rec.record_ref = None
        continue

    try:
        self.env[rec.res_model].browse(rec.res_id).check_access_rule("read")
        ...
```

what do you think about this?